src/mlte/store/store.py

```diff
@@ -0,0 +1,325 @@
+        :param model_id: The identifier for the model
+        :return: The model model
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def list_models(self, namespace_id: str) -> list[str]:
+        """
+        List all MLTE models in the given namespace.
+        :param namespace_id: The identifier for the namespace
+        :return: A collection of identifiers for all MLTE models
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def delete_model(self, namespace_id: str, model_id: str) -> Model:
+        """
+        Delete a MLTE model.
+        :param namespace_id: The identifier for the namespace
+        :param model_id: The identifier for the model
+        :return: The deleted model
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def create_version(
+        self, namespace_id: str, model_id: str, version: VersionCreate
+    ) -> Version:
+        """
+        Create a MLTE model version.
+        :param namespace_id: The identifier for the namespace
+        :param model_id: The identifier for the model
+        :param version: The version create model
+        :return: The created version
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def list_versions(self, namespace_id: str, model_id: str) -> list[str]:
+        """
+        List all MLTE versions in the given namespace and model.
+        :param namespace_id: the identifier for the namespace
+        :param model_id: The identifier for the model
+        :return: A collection of identifiers for all MLTE versions
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def read_version(
+        self, namespace_id: str, model_id: str, version_id: str
+    ) -> Version:
+        """
+        Read a MLTE model version.
+        :param namespace_id: The identifier for the namespace
+        :param model_id: The identifier for the model
+        :param version_id: The identifier for the model version
+        :return: The version model
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def delete_version(
+        self, namespace_id: str, model_id: str, version_id: str
+    ) -> Version:
+        """
+        Delete a MLTE model version.
+        :param namespace_id: The identifier for the namespace
+        :param model_id: The identifier for the model
+        :param version_id: The identifier for the model version
+        :return: The deleted version
+        """
+        raise NotImplementedError(
+            "Cannot invoke method on abstract StoreSession."
+        )
+
+    def write_negotiation_card(
```

Comment:
I am a little confused about the NegotiationCard being here. Are all artifacts going to have methods inside StoreSession? If so, this won't scale, or at best it is going to get really long. Is that the idea, for all artifacts to have 3 methods in StoreSession? It doesn't seem appropriate . Or is NegotiationCard special in some way that requires it to be in here? It is not part of the session, it looks like just another artifact to me.

Reply:
Yes, the idea in the current implementation is to have 3 methods within `StoreSession` (and each underlying implementation, actually) for each artifact type.

Originally, I intended the interface to include only a 3 methods for artifacts total, by operating on an abstract `ArtifactModel` instead of the individual artifact types themselves. I will revisit this to determine if I can make it more generic and thus scalable.

Reply:
Refactored this implementation. See updates.